src/components/tile-select/tile-select.test.js

```diff
@@ -0,0 +1,641 @@
+  const [isChecked, setIsChecked] = React.useState(false);
+  const [expanded, setExpanded] = React.useState(true);
+  const handleChange = (e) => {
+    const { value } = e.target;
+    setIsChecked(value === null ? false : true);
+  };
+  return (
+    <TileSelect
+      value="1"
+      id="single-1"
+      aria-label="single-1"
+      name="single"
+      title="Title"
+      subtitle="Subtitle"
+      description="Short and descriptive description"
+      checked={isChecked}
+      onChange={handleChange}
+      prefixAdornment={<Image height="40px" width="40px" />}
+      accordionContent={
+        <Box display="flex" flexWrap="wrap">
+          <Box flexGrow="1" pr={1}>
+            <Box
+              width="100%"
+              height="100px"
+              bg="primary"
+              display="inline-block"
+            />
+          </Box>
+          <Box flexGrow="1" pl={1}>
+            <Box
+              width="100%"
+              height="100px"
+              bg="primary"
+              display="inline-block"
+            />
+          </Box>
+        </Box>
+      }
+      accordionControl={(controlId, contentId) => (
+        <Button
+          buttonType="tertiary"
+          iconPosition="before"
+          iconType="chevron_down"
+          data-element="accordion-button"
+          onClick={() => setExpanded((expandedState) => !expandedState)}
+          px={1}
+          mt={2}
+          aria-controls={contentId}
+          id={controlId}
+        >
+          {expanded ? "Close" : "Open"} accordion
+        </Button>
+      )}
+      accordionExpanded={expanded}
+      setAccordionExpanded={setExpanded}
+      {...props}
+    />
+  );
+};
+
+const ActionButtonAdornment = ({ ...props }) => {
+  const [value, setValue] = React.useState(null);
+  return (
+    <TileSelectGroup
+      name="Tile Select"
+      value={value}
+      legend="Tile Select"
+      description="Pick one of the available options"
+      onChange={(e) => setValue(e.target.value)}
+    >
+      <TileSelect
+        value="2"
+        id="2"
+        aria-label="2"
+        title="Title"
+        subtitle="Subtitle"
+        titleAdornment={
+          <Icon
+            type="info"
+            tooltipMessage="Short and non descriptive message"
+          />
+        }
+        description="Short and descriptive description"
+        customActionButton={(onClick) => (
+          <Button
+            onClick={onClick}
+            buttonType="tertiary"
+            type="button"
+            px={1}
+            size="small"
+            destructive
+            disabled
+          >
+            Remove
+          </Button>
+        )}
+        actionButtonAdornment={
+          <Icon
+            type="info"
+            tooltipMessage="This tile cannot be removed at this time"
+          />
+        }
+        {...props}
+      />
+    </TileSelectGroup>
+  );
+};
+const TileSelectGroupComponent = ({ ...props }) => {
```

Comment:
why we need to create one more `group` component?

Reply:
There are few props which belongs to TileSelectGroup that's what I made a separate component for TileSelectGroup.